React2/week1/my-app/src/app/pages/form/page.jsx

```diff
@@ -0,0 +1,111 @@
+"use client"
+
+import {useRef, useState} from "react";
+import {useRouter} from "next/navigation";
+import Navbar from "@/app/pages/components/navbar";
+
+export default function FormPage() {
+    const firstNameRef = useRef(null);
+    const lastNameRef = useRef(null);
+    const phoneRef = useRef(null);
+    const emailRef = useRef(null);
+    const [errors, setErrors] = useState({});
+    const router = useRouter();
+
+    // Function to handle focus on the next input field
+    const focusNextField = (currentField) => {
+        if (currentField === firstNameRef.current) lastNameRef.current.focus();
+        else if (currentField === lastNameRef.current) phoneRef.current.focus();
+        else if (currentField === phoneRef.current) emailRef.current.focus();
+    };
+
+    // Function to validate input fields
+    const validateFields = () => {
+        const newErrors = {};
+        if (!firstNameRef.current.value) newErrors.firstName = "First name is required";
+        if (!lastNameRef.current.value) newErrors.lastName = "Last name is required";
+        if (!phoneRef.current.value.match(/^\d+$/)) newErrors.phone = "Phone number is invalid";
+        if (!emailRef.current.value.match(/^\S+@\S+\.\S+$/)) newErrors.email = "Email is invalid";
+        setErrors(newErrors);
+
+        return Object.keys(newErrors).length === 0;
+    };
```

Comment:
After I click "submit", there is still no errors shown in the form. Can you explain why?

<img width="532" alt="image" src="https://github.com/user-attachments/assets/79e2a242-b262-44a6-b3b5-5947d49b488f">

Reply:
Because I didn't use Material-UI components for this. I already changed it. Thanks